server/apps/subscriptions.go

```diff
@@ -4,31 +4,149 @@
 package apps
 
 import (
+	pluginapi "github.com/mattermost/mattermost-plugin-api"
 	"github.com/mattermost/mattermost-plugin-apps/server/configurator"
+	"github.com/pkg/errors"
 )
 
+const SubsPrefixKey = "sub_"
+
 type Subscriptions interface {
-	GetSubscriptionsForChannel(subj SubscriptionSubject, channelID string) ([]*Subscription, error)
+	GetSubscriptionsForChannelOrTeam(subj SubscriptionSubject, channelOrTeamID string) ([]*Subscription, error)
```

Comment:
- 1/5 s/`Subscriptions`/`Subs`/g, use `subs *Subs` for variable name, consistently.
- 1/5 Keep `Subscription`. Reason: typography.
- 2/5 use `sub *Subscription` except in obvious context (member functions) `s *Subscription` is acceptable.
- 2/5 `GetHookSubs(..., channelID, teamID, etc...)` and `GetAppSubs`(appID)`

Reply:
@levb, what is your take on `GetSubsForChannelOrTeam` vs `GetChannelOrTeamSubs`?

Reply:
For each of the notification structs.  Do you prefer the shortened field name `SubsID`?
```go
type SubscriptionDeletedNotification struct {
	SubscriptionID SubscriptionID
	Subject        SubscriptionSubject
	Expanded       *Expanded
}

```

Reply:
See [`036f2bb` (#9)](https://github.com/mattermost/mattermost-plugin-apps/pull/9/commits/036f2bb0eb151409ace05a2cd5b2e06d38813f6e)